Doc/whatsnew/3.15.rst

```diff
@@ -1068,6 +1068,10 @@ Deprecated C APIs
   since 3.15 and will be removed in 3.17.
   (Contributed by Nikita Sobolev in :gh:`136355`.)
 
+* The :c:macro:`!Py_INFINITY` macro is :term:`soft deprecated`,
+  use the C standard :c:macro:`!INFINITY` instead.
```

Comment:
You should mention the header providing it. I'm not sure if we should mention that C11 is required to get the macro. (C99 in practice.)

```suggestion
  use the C11 standard ``<math.h>`` :c:macro:`!INFINITY` instead.
```